pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1044,6 +1047,11 @@ class Timedelta(_Timedelta):
     __rsub__ = _binary_op_method_timedeltalike(lambda x, y: y - x, '__rsub__')
 
     def __mul__(self, other):
+        if is_integer_object(other) or is_float_object(other):
+            # includes numpy scalars that would otherwise be caught by dtype
+            # check below
```

Comment:
another option to do this would be `if hasattr(other, 'dtype') and not (is_integer_object(other) or is_float_object(other)):` 
that way you don't need to repeat the `Timedelta(other * self.value, unit='ns')` twice

Reply:
Good idea.  I like having if/elif branches that are explicitly mutually explicit and so don't rely on the ordering.